src/ios/SimpleCameraPreview.m

```diff
@@ -203,11 +203,18 @@ - (void) capture{
     [self.sessionManager.stillImageOutput captureStillImageAsynchronouslyFromConnection:connection completionHandler:^(CMSampleBufferRef sampleBuffer, NSError *error) {
         if (error) {
             NSLog(@"%@", error);
-            NSString* errorDescription =  error.description ? error.description : @"";
-            errorDescription = [@"Error taking picture: " stringByAppendingString:errorDescription];
-            CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:errorDescription];
+            CDVPluginResult *pluginResult;
+            if (!CMSampleBufferIsValid(sampleBuffer)) {
+                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Sample buffer not valid"];
+            } else {
+                NSString* errorDescription =  error.description ? error.description : @"";
+                errorDescription = [@"Error taking picture: " stringByAppendingString:errorDescription];
+                pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:errorDescription];
+            }
             [self.commandDelegate sendPluginResult:pluginResult callbackId:self.onPictureTakenHandlerId];
-        } else {
+            return;
+        }
+        [self runBlockWithTryCatch:^{
```

Comment:
Is there a reason not to put the try catch inline here ?

Reply:
@ombr We added this in case for example if `sampleBuffer` is corrupted while creating the image data. That's why we included this here.